### ISSUE TITLE
Twister: Make tests run in a defined sequence when accessing same hardware resource

### ISSUE THREAD

**Issue by nashif** [zephyrproject-rtos/zephyr#28316]
**Is your enhancement proposal related to a problem? Please describe.**

When tests are accessing a resource that cant be shared (see #25295), we should run tests in sequence and block. running in parallel will result in concurrency and failed tests.

**Describe the solution you'd like**
Make tests block when accessing a common and share resource, do not let them run in parallel.

**Comment by zephyrbot** [comment]
Hi @nashif,

This issue, marked as an Enhancement, was opened a while ago and did not get any traction. It was just assigned to you based on the labels. If you don't consider yourself the right person to address this issue, please re-assing it to the right person.

Please take a moment to review if the issue is still relevant to the project.  If it is, please provide feedback and direction on how to move forward. If it is not, has already been addressed, is a duplicate, or is no longer relevant, please close it with a short comment explaining the reason.

@nashif you are also encouraged to help moving this issue forward by providing additional information and confirming this request/issue is still relevant to you.

Thanks!

**Comment by nashif** [comment]
did not see the issue for a while, closing
